### ISSUE TITLE
Training fails after 1 epoch, kills terminal

### ISSUE THREAD

**Issue by aenoca** [ultralytics/ultralytics#21606]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Training fails and kills the terminal after the first epoch.

### Environment

```
Ultralytics 8.3.171 🚀 Python-3.10.12 torch-2.5.1+cu124 CUDA:0 (NVIDIA GeForce RTX 4090, 24098MiB)
Setup complete ✅ (32 CPUs, 62.6 GB RAM, 1514.7/1599.5 GB disk)

OS                  Linux-6.8.0-65-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.12
Install             pip
Path                /home/anoca/.local/lib/python3.10/site-packages/ultralytics
RAM                 62.64 GB
Disk                1514.7/1599.5 GB
CPU                 Intel Core(TM) i9-14900KF
CPU count           32
GPU                 NVIDIA GeForce RTX 4090, 24098MiB
GPU count           1
CUDA                12.4

numpy               ✅ 1.26.4>=1.23.0
matplotlib          ✅ 3.10.0>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.12.0>=1.4.1
torch               ✅ 2.5.1>=1.8.0
torch               ✅ 2.5.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1>=0.9.0
tqdm                ✅ 4.66.5>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
ultralytics-thop    ✅ 2.0.14>=2.0.0
```

### Minimal Reproducible Example

```
from ultralytics import YOLO
import yaml


import wandb
from wandb.integration.ultralytics import add_wandb_callback

# model = YOLO("yolo11n.pt")  # load a pretrained model (recommended for training)
# results = model.train(data="coco.yaml",
#                       epochs=100, 
#                       imgsz=160,
#                       batch=16,  # adjust batch size based on your GPU memory
#                       device="0",  # specify the GPU device, e.g., "0" for the first GPU
#                       name="yolo11n_160x160",  # name of
#                       )

# load the classes that we actually want to train on
with open("subselected_classes.yaml", "r") as f:
    subselected_classes = yaml.safe_load(f)

selected_classes = [cls['id'] for cls in subselected_classes['classes']]
print(f"Selected classes for training: {selected_classes}")
print(type(selected_classes[0]))


# wandb.init(project="ultralytics", job_type="train")

# Load a model
model = YOLO("yolo11n.pt")  # load a pretrained model (recommended for training)

# add_wandb_callback(model, enable_model_checkpointing=True)

# Train the model
results = model.train(data="object365.yaml", 
                    epochs=100, 
                    imgsz=160,
                    classes=selected_classes,  # use the selected classes
                    batch=128,  # adjust batch size based on your GPU memory
                    device="0",  # specify the GPU device, e.g., "0" for the first GPU
                    project="ultralytics",  # project name in wandb
                    name="yolo11n_160x160",  # name of
                    warmup_epochs=0,  # no warmup epochs,
                    workers=16,
                    fraction=0.1
                    )
```

subselected_classes.yaml
```
classes:
  - id: 0
    name: Person
  - id: 5
    name: Car
  - id: 21
    name: Boat
  - id: 34
    name: SUV
  - id: 49
    name: Van
  - id: 55
    name: Bus
  - id: 58
    name: Motorcycle
  - id: 65
    name: Truck
  - id: 87
    name: Pickup Truck
  - id: 114
    name: Airplane
  - id: 116
    name: Train
  - id: 126
    name: Sports Car
  - id: 131
    name: Crane
  # - id: 138
  #   name: Surveillance Camera
  - id: 147
    name: Gun
  - id: 188
    name: Fire Truck
  - id: 211
    name: Ship
  - id: 248
    name: Ambulance
  - id: 263
    name: Helicopter
  - id: 109
    name: Machinery Vehicle
  - id: 228
    name: Hot-air balloon
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @aenoca, thank you for reporting this issue and for providing detailed environment info and code! 🚀 This is an automated response to help you get faster support—an Ultralytics engineer will review and assist you soon.

Since this appears to be a 🐛 Bug Report, please ensure your [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) is as concise as possible and includes only the code and files necessary to reproduce the problem. If you can further reduce your example or provide any relevant logs or error messages, that will help us debug more efficiently.

If you are new to Ultralytics or need reference, check the [Docs](https://docs.ultralytics.com/), where you can find many helpful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.

Join the Ultralytics community for support and discussion:
- [Discord](https://discord.com/invite/ultralytics) 🎧 (real-time chat)
- [Discourse](https://community.ultralytics.com/) (in-depth discussions)
- [Subreddit](https://www.reddit.com/r/Ultralytics/) (community threads)

## Upgrade

Please upgrade to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the most recent version:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and contribution!

**Comment by Y-T-G** [comment]
Can you post the training log?
